ppa/archive/util.py

```diff
@@ -139,12 +174,22 @@ def add_items(self, log_msg_src=None, user=None):
         # unsuppressed work; perhaps we could do a similar check here?
 
         for htid in self.htids:
+            # if rsync did not create the expected directory,
+            # set error code and bail out
+            expected_path = os.path.join(settings.HATHI_DATA,
+                                         self.pairtree_paths[htid])
+            if not os.path.isdir(expected_path):
+                self.results[htid] = self.RSYNC_ERROR
+                continue
```

Comment:
Right, I hadn't realized that you wanted to check that each new ID had properly synced. This looks good 🙂